src/ApiRequest/FetchReportApiRequest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+namespace Silktide\ProspectClient\ApiRequest;
+
+use Error;
+use Silktide\ProspectClient\ApiException\ReportStillRunningException;
+use Silktide\ProspectClient\ApiResponse\FetchReportApiResponse;
+use Silktide\ProspectClient\ApiResponse\AbstractApiResponse;
+use TypeError;
+
+class FetchReportApiRequest extends AbstractApiRequest
+{
+    /** @var string */
+    protected $apiPath = "report";
+    /** @var string */
+    protected $apiMethod = "get";
+
+    /** @var */
+    private $id;
+
+    /** @return FetchReportApiResponse */
+    public function execute(): AbstractApiResponse
+    {
+        try {
+            $this->apiPathSuffix = $this->id;
+            // TODO: Temporary fix while PHP 7.2 is a requirement:
+            if(!$this->id) {
```

Comment:
The developer only needs to pass the ID to `fetch` ( https://github.com/Silktide/prospect-php-client/blob/6e42e86d90a6367225a861e1eedd7476a489de87/src/Api/ReportApi.php#L28 ). The FetchReportApiRequest's constructor needs to have the HttpWrapper passed into it, but is only used internally so I'd say this is fine.

Reply:
Ideally, you want your request object to always be in a valid state.

WIth `create` for example, any request you create will be a valid request because the validation step will need to come from the server side. This is because we have the concept of website-less testing, but it's only enabled for some accounts.

In the case of `reanalyse` and `fetch`, the `$reportId` will always be needed for the request to be valid, as such ideally should be fed in on object initialisation.

This removes the need for a second validation step before execute is run and simplifies the code.